android-release-build.md

```diff
@@ -0,0 +1,159 @@
+
+When you're ready to prepare a *release* version for Android, for example to 
+[publish to the Google Play Store][play], follow the steps on this page.
+
+* TOC Placeholder
+{:toc}
+
+## Reviewing the App Manifest
+
+When you create a new app, a default [App Manifest][manifest] file is created
+at `<app dir>/android/src/main/AndroidManifest.xml`. Before releasing, 
+review the contents of this file and verify the values are correct, especially:
+
+* `application`: Edit the [`application`][applicationtag] tag to reflect the final name of the app.
+
+* `manifest`: Specify the [version][versions] code and name in the [`manifest`][manifesttag] tag.
+
+* `uses-sdk`: Specify the [API versions][apiversionstag] the app is compatible with.
+
+* `uses-permission`: Remove the `android.permission.INTERNET` [permission][permissiontag] if 
+  your application code does not need Internet access. The standard template includes this tag 
+  to enable communication between Flutter tools and a running app.  
+
+## Adding a Launcher icon
+
+When a new Flutter app is created, it has a default Launcher icon. To
+customize this icon:
+
+1. Review the [Android Launcher Icons][launchericons] guidelines for icon
+design.
+
+1. In the `<app dir>/android/src/main/res/` directory, place your icon files
+in folders named using [Configuration Qualifiers][configurationqualifiers].
+The default `mipmap-` folders demonstrate the correct naming convention.
+
+1. In `AndroidManifest.xml`, update the [`application`][applicationtag] tag's
+`android:icon` attribute to reference icons from the previous step (e.g.
+`<application android:icon="@mipmap/ic_launcher" ...`).
+
+1. To verify the icon has been replaced, run your app using `flutter run` 
+and inspect the app icon in the Launcher.
+
+## Signing the app
+
+### Create a keystore
+If you have an existing keystore, skip to the next step. If not, create one
+by running the following at the command line:
+`keytool -genkey -v -keystore ~/key.jks -keyalg RSA -keysize 2048 -validity 10000 -alias key`
+
+*Note*: Keep this file private; do not check it into public source control.
+
+### Reference the keystore from the app
+
+Create a file named `<app dir>/android/key.properties` that contains a
+reference to your keystore:
+
+```
+storePassword=<password from previous step>
+keyPassword=<password from previous step>
+keyAlias=key
+storeFile=<location of the key store file, e.g. '/Users/<user name>/key.jks'>
+```
+
+*Note*: Keep this file private; do not check it into public source control.
+
+### Configure signing in gradle
+
+Configure signing for your app by editing the `<app dir>/android/app/build.gradle` 
+file.
+
+1. Replace:
+```
+   android {
+```
+   with the keystore information from your properties file:
+```
+   def keystorePropertiesFile = rootProject.file("key.properties")
+   def keystoreProperties = new Properties()
+   keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
```

Comment:
Tiny nit: This works, and is how most examples show how to do it, but it also leaves the properties file open until the build is done (or a GC occurs and the finalizers run). A version that closes the stream would look like:

```groovy
def keystoreProperties = new Properties()
rootProject.file('key.properties').withInputStream { stream ->
  keystoreProperties.load(stream)
}
```

But it's also fine to leave as is.